### PR TITLE
HPCC-18218 Upgrade ESP to timeout session after client

### PR DESCRIPTION
**Pull request by wangkx**
The existing ESP/ECLwatch controls session timeout using the same timeout value. 
It is difficult to make sure a session to be timed out on both sides at the same time 
due to network delay or different clock times on ESP server and ECLwatch client 
box. In this fix, two timeout values, one for client side and one for server side, are
used to controls session timeout. ECLwatch or other ESP client may use the timeout 
value for client side to force a session timeout. ESP will clean the session based on 
the timeout value for server side. The code forces that the timeout value for server 
side is not less than the timeout value for client side.

Signed-off-by: wangkx <kevin.wang@lexisnexis.com>

## Type of change:
- [ ] This change is a bug fix (non-breaking change which fixes an issue).
- [x] This change is a new feature (non-breaking change which adds functionality).
- [ ] This change improves the code (refactor or other change that does not change the functionality)
- [ ] This change fixes warnings (the fix does not alter the functionality or the generated code)
- [ ] This change is a breaking change (fix or feature that will cause existing behavior to change).
- [ ] This change alters the query API (existing queries will have to be recompiled)

## Checklist:
- [x] My code follows the code style of this project.
  - [x] My code does not create any new warnings from compiler, build system, or lint.
- [x] The commit message is properly formatted and free of typos.
  - [ ] The commit message title makes sense in a changelog, by itself.
  - [x] The commit is signed.
- [ ] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly, or...
  - [ ] I have created a JIRA ticket to update the documentation.
  - [ ] Any new interfaces or exported functions are appropriately commented.
- [x] I have read the CONTRIBUTORS document.
- [x] The change has been fully tested:
  - [ ] I have added tests to cover my changes.
  - [ ] All new and existing tests passed.
  - [ ] I have checked that this change does not introduce memory leaks.
  - [ ] I have used Valgrind or similar tools to check for potential issues.
- [x] I have given due consideration to all of the following potential concerns:
  - [ ] Scalability
  - [ ] Performance
  - [ ] Security
  - [ ] Thread-safety
  - [ ] Premature optimization
  - [ ] Existing deployed queries will not be broken
  - [ ] This change fixes the problem, not just the symptom
  - [ ] The target branch of this pull request is appropriate for such a change.
- [ ] There are no similar instances of the same problem that should be addressed
  - [ ] I have addressed them here
  - [ ] I have raised JIRA issues to address them separately
- [ ] This is a user interface / front-end modification
  - [ ] I have tested my changes in multiple modern browsers
  - [ ] The component(s) render as expected

### CODE REVIEW

**Comment by wangkx** [comment]
@RussWhitehead please review.

**Comment by RussWhitehead** [comment]
Please add a commit message to describe the reason for this PR

File: esp/bindings/http/platform/httpbinding.cpp [hunk 1]
```diff
@@ -349,13 +349,23 @@ void EspHttpBinding::readAuthDomainCfg(IPropertyTree* procCfg)
...
         }
 
-        int sessionTimeoutMinutes = authDomainTree->getPropInt("@sessionTimeoutMinutes", 0);
-        if (sessionTimeoutMinutes == 0)
-            sessionTimeoutSeconds = ESP_SESSION_TIMEOUT;
-        else if (sessionTimeoutMinutes < 0)
-            sessionTimeoutSeconds = -1;
+        int clientSessionTimeoutMinutes = authDomainTree->getPropInt("@clientSessionTimeoutMinutes", 0);
+        if (clientSessionTimeoutMinutes == 0)
+            clientSessionTimeoutSeconds = ESP_SESSION_TIMEOUT;
```
Comment (RussWhitehead) [inline esp/bindings/http/platform/httpbinding.cpp]: Can't ESP_SESSION_TIMEOUT be passed as the second argument to getPropInt?

File: esp/bindings/http/platform/httpbinding.hpp [hunk 2]
```diff
@@ -164,8 +164,9 @@ class esp_http_decl EspHttpBinding :
 
     StringAttr              loginURL;
     StringAttr              logoutURL;
-    int                     sessionTimeoutSeconds = ESP_SESSION_TIMEOUT; //-1: never
-    int                     checkSessionTimeoutSeconds = ESP_CHECK_SESSION_TIMEOUT;
+    int                     serverSessionTimeoutSeconds = 2*ESP_SESSION_TIMEOUT; //-1: never
```
Comment (RussWhitehead) [inline esp/bindings/http/platform/httpbinding.hpp]: Would it be helpful to create a "#define NEVER_TIMEOUT -1"  ?

File: initfiles/componentfiles/configxml/esp.xsd.in [hunk 3]
```diff
@@ -415,20 +415,29 @@
 	                </xs:appinfo>
 	              </xs:annotation>
 	            </xs:attribute>
-	            <xs:attribute name="sessionTimeoutMinutes" use="optional" default="120">
+	            <xs:attribute name="clientSessionTimeoutMinutes" use="optional" default="120">
 	              <xs:annotation>
 	                <xs:appinfo>
-	                  <tooltip>Inactive session duration, in minutes. Specify 0 for default timeout, -1 for never time out.</tooltip>
+	                  <tooltip>Inactive session duration, in minutes, that is used to timeout a sesion on ESP client. Specify 0 for default timeout, -1 for never time out.</tooltip>
```
Comment (RussWhitehead) [inline initfiles/componentfiles/configxml/esp.xsd.in]: "session" misspelled

**Review by RussWhitehead — CHANGES_REQUESTED** [review]
A couple minor questions/comments

**Comment by RussWhitehead** [comment]
@wangkx Looks good

**Comment by wangkx** [comment]
@richardkchapman this PR is ready to be merged.

**Comment by richardkchapman** [comment]
@wangkx Please squash

**Comment by wangkx** [comment]
@richardkchapman this PR has been squashed. Please merge.

**Pull request merged**